core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -129,9 +139,20 @@ public void cancelled(Context context) {
     return this;
   }
 
+  ClientCallImpl<ReqT, RespT> setCompressorRegistry(CompressorRegistry compressorRegistry) {
+    this.compressorRegistry = compressorRegistry;
+    return this;
+  }
+
+  ClientCallImpl<ReqT, RespT> setKnownMessageEncodings(Set<String> knownMessageEncodings) {
```

Comment:
Methods named like this usually imply the method would only consume the content of the given container, while this method will keep the container and the class will modify it later. To make this clear, I would rename it to `setKnownMessageEncodingRegistry()`.